src/main/java/appeng/parts/p2p/ItemP2PTunnelPart.java

```diff
@@ -118,36 +84,43 @@ public ItemStack insertItem(int slot, @Nonnull ItemStack stack, boolean simulate
                 int overflow = amountPerOutput == 0 ? amount : amount % amountPerOutput;
 
                 for (ItemP2PTunnelPart target : ItemP2PTunnelPart.this.getOutputs()) {
-                    final IItemHandler output = target.getAttachedItemHandler();
-                    final int toSend = amountPerOutput + overflow;
-                    final ItemStack fillWithItemStack = stack.copy();
-                    fillWithItemStack.setCount(toSend);
-
-                    ItemStack received = ItemStack.EMPTY;
+                    try (CapabilityGuard capabilityGuard = target.getAdjacentCapability()) {
+                        final IItemHandler output = capabilityGuard.get();
+                        final int toSend = amountPerOutput + overflow;
 
-                    for (int i = 0; i < output.getSlots(); i++) {
-                        received = output.insertItem(i, fillWithItemStack, simulate);
-
-                        if (received.isEmpty()) {
+                        if (toSend <= 0) {
```

Comment:
If `toSend` is actually negative here, would that not indicate an error and potentially shouldn't be swallowed?

Reply:
That's an error in another mod, yes. I'm not sure how hard we should try to catch it? We can print the handler itself, the adjacent tile, the adjacent blockstate, etc... Not sure if that's worth it?

Reply:
We can check that `0 <= sent <= toSend`, what do you think?

Reply:
Forge's helpers don't check this, so it will already fail inside `ItemHandlerHelper#insertItem`. I don't think we should spend too much energy on that.

Reply:
It's no use, it will fail in `ItemHandlerHelper#insertItem`.